File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) -1
        while left <= right:
            m = (left + right) // 2
            print(m)
            if nums[m] == target:
                return m
            if nums[m] > target:
                right = m - 1
            if nums[m] < target:
                left = m + 1
            print(left, right)

        return -1

File: Data Structures & Algorithms/remove-element/submission-0.java
class Solution {
    public int removeElement(int[] nums, int val) {
        Arrays.sort(nums);
        int right = nums.length - 1;
        int left = 0;
        while (left <= right){
            if (nums[right] == val) {
                right -= 1;
                continue;
            } else if (nums[left] == val) {
                int temp = nums[right];
                nums[right] = nums[left];
                nums[left] = temp;
                right -= 1;
            }
            left += 1;
        }
        return right+1;
    }
}